api/handlers/asset_handler.go

```diff
@@ -134,7 +134,67 @@ func (h *AssetHandler) Upsert(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := h.saveLineage(r.Context(), payload); err != nil {
+	if err := h.saveLineage(r.Context(), ast, payload.Upstreams, payload.Downstreams); err != nil {
+		internalServerError(w, h.logger, err.Error())
+		return
+	}
+
+	writeJSON(w, http.StatusOK, map[string]interface{}{
+		"id": ast.ID,
+	})
+}
+
+func (h *AssetHandler) UpsertPatch(w http.ResponseWriter, r *http.Request) {
```

Comment:
Should we just say patch? to avoid confusion

Reply:
was thinking about that, but this also allows creation.